### ISSUE TITLE
Logging documentation could explain log observers a bit better

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4642]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#4642|
|Type|enhancement|
|Created|2010-08-26 18:56:06Z|
|Branch|https://github.com/twisted/twisted/tree/logging-docs-4642|

1. The logging howto should do a better job explaining that you only need a custom LogObserver if you want custom logging handling -- currently it just says "Log observers are the basis of the Twisted logging system." We should make it clearer that log.msg dispatches to a list of registered observers, Twisted includes a default file observer, and writing a custom observer is usually not necessary.

2. Regarding the Python logging module, the language is very ambiguous, e.g. the phrase "If your application uses the logging module". If you don't know that Python stdlib has a logging.py the whole section will be confusing. The text should be more explicit about the fact it is referring to the Python standard library module logging.py, and also include a link.

<details><summary>Searchable metadata</summary>

```
trac-id__4642 4642
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__ 
branch__branches_logging_docs_4642 branches/logging-docs-4642
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy_documentation easy documentation
time__1282848966000000 1282848966000000
changetime__1286123336000000 1286123336000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph removed owner|
|-|-|

+1

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [30073]) Branching to 'logging-docs-4642'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [30074]) Talk about how there's a list of observers

refs [#4642](https://github.com/twisted/twisted/issues/4642)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [30075]) Say "standard library logging module" everywhere

refs [#4642](https://github.com/twisted/twisted/issues/4642)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/logging-docs-4642)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess set owner to @exarkun|
|-|-|

Thanks for working on this, exarkun. Some things unrelated to your changes that I noticed in the file:

 * "Unless logLevel is provided" ==> "Unless <code>logLevel</code> is provided"
 * "some handlers (e.g. SMTP, HTTP) uses network so" ==> "some handlers (e.g. SMTP, HTTP) use the network and so"
 * "the bridge": what "the bridge" means hasn't been introduced before and is potentially confusing.
 * "Whenever an <code>log.msg</code>" ==> "Whenever a <code>log.msg</code>"
 * The hyperlinked "msg" 404s.

Beyond those things and a news file, your changes look great and ready to merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [30093]) Fix language issues pointed out in the review

refs [#4642](https://github.com/twisted/twisted/issues/4642)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

Thanks for the review.  The first four points fixed in the above commit.  The last point is [#11366](https://github.com/twisted/twisted/issues/11366).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [30095]) Merge logging-docs-4642

Author: exarkun
Reviewer: jesstess
Fixes: [#4642](https://github.com/twisted/twisted/issues/4642)

Clarify the language in the logging howto with respect to the standard library
logging module.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
